scripts/prepare_docker_buildinfo.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+IMAGENAME=$1
+DOCKERFILE=$2
+ARCH=$3
+DOCKERFILE_TARGE=$4
+DISTRO=$5
+
+[ -z "$BUILD_SLAVE" ] && BUILD_SLAVE=n
+[ -z "$DOCKERFILE_TARGE" ] && DOCKERFILE_TARGE=$DOCKERFILE
+DOCKERFILE_PATH=$(dirname "$DOCKERFILE_TARGE")
+BUILDINFO_PATH="${DOCKERFILE_PATH}/buildinfo"
+BUILDINFO_VERSION_PATH="${BUILDINFO_PATH}/versions"
+
+[ -d $BUILDINFO_PATH ] && rm -rf $BUILDINFO_PATH
+mkdir -p $BUILDINFO_VERSION_PATH
+
+# Get the debian distribution from the docker base image
+if [ -z "$DISTRO" ]; then
+    DOCKER_BASE_IMAGE=$(grep "^FROM" $DOCKERFILE | head -n 1 | awk '{print $2}')
+    DISTRO=$(docker run --rm --entrypoint "" $DOCKER_BASE_IMAGE cat /etc/os-release | grep VERSION_CODENAME | cut -d= -f2)
+    [ -z "$DISTRO" ] && DISTRO=jessie
+fi
+
+DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo 
+COPY ["buildinfo", "/usr/local/share/buildinfo"]
+RUN dpkg -i /usr/local/share/buildinfo/sonic-build-tools_1.0_all.deb
+RUN apt-mark hold sonic-build-tools
+RUN pre_run_buildinfo'
+
+# Add the auto-generate code if it is not added in the target Dockerfile
+if [ ! -f $DOCKERFILE_TARGE ] || ! grep -q "Auto-Generated for buildinfo" $DOCKERFILE_TARGE; then
+    # Insert the docker build script before the RUN command
+    LINE_NUMBER=$(grep -Fn -m 1 'RUN' $DOCKERFILE | cut -d: -f1)
+    TEMP_FILE=$(mktemp)
+    awk -v text="${DOCKERFILE_PRE_SCRIPT}" -v linenumber=$LINE_NUMBER 'NR==linenumber{print text}1' $DOCKERFILE > $TEMP_FILE
+
+    # Append the docker build script at the end of the docker file
+    echo "RUN post_run_buildinfo" >> $TEMP_FILE
+    [ "$BUILD_SLAVE" != "y" ] && echo "RUN set_build_hooks -d" >> $TEMP_FILE
+
+    cat $TEMP_FILE > $DOCKERFILE_TARGE
+    rm -f $TEMP_FILE
+fi
+
+# Copy the build info config
+cp -rf files/build/buildinfo/* $BUILDINFO_PATH
+
+# Build the slave running config
+if [ "$BUILD_SLAVE" == "y" ]; then
+    scripts/versions_manager.py generate -t "${BUILDINFO_PATH}/build/versions" -n "build-${IMAGENAME}" -d "$DISTRO" -a "$ARCH"
+    touch ${BUILDINFO_PATH}/build/versions/versions-deb
+fi
```

Comment:
Move this section into prepare_slave_container_build_info

Reply:
Offline discussed, should remove L48~L52.

---
In reply to: [538023392](https://github.com/Azure/sonic-buildimage/pull/5718#discussion_r538023392) [](ancestors = 538023392)